bin/scripts/continious-data-population.js

```diff
@@ -0,0 +1,1861 @@
+
+            if (Math.random() >= 0.8) {
+                events = events.concat(this.getScrollmapEvent(appKey, templateId));
+            }
+        }
+
+        return events;
+    };
+
+    this.getScrollmapEvent = function(appKey, templateId) {
+        var populatorType = null;
+        if (templateId.length <= 10) { //defaultTemplate
+            defaultTemplates.forEach(function(template) {
+                if (template.name === templateId) {
+                    populatorType = template._id;
+                }
+            });
+        }
+        else {
+            populatorType = template._id;
+        }
+        var views = ["/populator/" + appKey + "/demo-" + populatorType + ".html"];
+        var event = {
+            "key": "[CLY]_action",
+            "count": 1,
+            "timestamp": this.ts,
+            "hour": getRandomInt(0, 23),
+            "dow": getRandomInt(0, 6),
+            "test": 1
+        };
+        this.ts += 1000;
+        event.segmentation = {};
+        event.segmentation.type = "scroll";
+        event.segmentation.y = getRandomInt(0, 3602) + 983;
+        event.segmentation.width = 1440;
+        event.segmentation.height = 3586;
+        event.domain = SERVER_URL;
+        event.segmentation.view = views[Math.floor(Math.random() * views.length)];
+        return [event];
+    };
+
+    this.addEvent = function(template, app, templateId) {
+        var req = {};
+        var events;
+
+        if (!this.isRegistered) {
+            this.isRegistered = true;
+            events = this.getEvent("[CLY]_view", template && template.events && template.events["[CLY]_view"], app.key, templateId).concat(this.getEvent("[CLY]_orientation", template && template.events && template.events["[CLY]_orientation"], app.key, templateId), this.getEvents(4, template && template.events, app.key, templateId));
+            // events = events.sort(() => 0.5 - Math.random()).slice(eventChance);
+            req = {timestamp: this.ts, events: events};
+            req.events = req.events.concat(this.getHeatmapEvents(app.key, templateId));
+            req.events = req.events.concat(this.getFeedbackEvents(app._id));
+            req.events = req.events.concat(this.getScrollmapEvents(app.key, templateId));
+            req.events = req.events[Math.floor(Math.random() * req.events.length)];
+        }
+        else {
+            events = this.getEvent("[CLY]_view", template && template.events && template.events["[CLY]_view"], app.key, templateId).concat(this.getEvent("[CLY]_orientation", template && template.events && template.events["[CLY]_orientation"], app.key, templateId), this.getEvents(4, template && template.events, app.key, templateId));
+            // events = events.sort(() => 0.5 - Math.random()).slice(eventChance);
+            events = events[Math.floor(Math.random() * events.length)];
+            req = {timestamp: this.ts, events: events};
+        }
+        this.request(req, app.key);
```

Comment:
req.events.length === events count you are adding to session. 
So:
 users[inputs[z].APP_ID][index].eventsInSession+=req.events.length;
 (You could return that value and then sum OR pass  users[inputs[z].APP_ID][index] and add there.
 (Objects are passed by reference)